File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/resolve/ResolvePackagingsTest.java
package org.jetbrains.plugins.scala.lang.resolve;

import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiReference;
import org.jetbrains.plugins.scala.lang.psi.api.base.ScPrimaryConstructor;

/**
 * @author ilyas
 */
public class ResolvePackagingsTest extends ScalaResolveTestCase {
  public String folderPath() {
    return super.folderPath() + "resolve/packages/solid/my/scala/stuff/";
  }

  @Override
  protected String rootPath() {
    return super.folderPath() + "resolve/packages/";
  }

  public void testMain() throws Exception {
    PsiReference ref = findReferenceAtCaret();
    final PsiElement psiElement = ref.resolve();
    assertTrue(psiElement instanceof ScPrimaryConstructor);
  }

}

File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/resolve/ResolvePackagingsTest.scala
package org.jetbrains.plugins.scala.lang.resolve

import org.jetbrains.plugins.scala.extensions.{ObjectExt, PathExt}
import org.jetbrains.plugins.scala.lang.psi.api.base.ScPrimaryConstructor
import org.junit.Assert._

import java.nio.file.Path

class ResolvePackagingsTest extends ScalaResolveTestCase {
  override def folderPath: Path = super.folderPath / "resolve" / "packages" / "solid" / "my" / "scala" / "stuff"

  override protected def sourceRootPath: Path = super.folderPath / "resolve" / "packages"

  def testMain(): Unit = {
    val ref = findReferenceAtCaret()
    val psiElement = ref.resolve
    assertTrue(psiElement.is[ScPrimaryConstructor])
  }
}
